backend/src/books/books.controller.ts

```diff
@@ -0,0 +1,14 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { BooksService } from './books.service';
+
+import { Book } from '../types';
+
+@Controller('books')
+export class BooksController {
+  constructor(private readonly bookService: BooksService) {}
+
+  @Get()
+  getBooks(@Query('query') query: string): Promise<Book[]> {
```

Comment:
Perhaps some [validation](https://docs.nestjs.com/pipes#validation-use-case) using class-validator could be helpful here.

Reply:
I decided to leave this one out at this point as each query parameter is parsed as a string. However, this will definitely become relevant at a later point